File: 0447-Number of Boomerangs/0447-Number of Boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<pair<int, int>>& points) {
        int count = 0;
        for (auto& p : points)
        {
            unordered_map<long, int> table;
            for (auto& q : points)
            {
                count += 2 * table[getDistance(p, q)]++;
            }
        }
        
        return count;
    }
    
    long getDistance(pair<int, int>& p1, pair<int, int>& p2)
    {
        return (p1.first - p2.first) * (p1.first - p2.first) + (p1.second - p2.second) * (p1.second - p2.second);
    }
};

File: 0401-0500/0447-Number of Boomerangs/0447-Number of Boomerangs.py
import collections
class Solution:
    def numberOfBoomerangs(self, points: List[List[int]]) -> int:
        count = 0
        for p in points:
            table = collections.Counter()
            for q in points:
                diff = (p[0] - q[0])**2 + (p[1] - q[1])**2
                count += table[diff]*2
                table[diff] += 1
        return count
